test/foundry/FulfillOrderTest.t.sol

```diff
@@ -510,6 +512,100 @@ contract FulfillOrderTest is BaseOrderTest {
         );
     }
 
+    function _getLengthAtOffsetInOrderCalldata(
+        bytes memory orderCalldata,
+        uint256 relativeLengthOffset
+    ) internal returns (uint256 length) {
+        assembly {
+            let absoluteLengthOffset := add(orderCalldata, relativeLengthOffset)
+            length := mload(absoluteLengthOffset)
+        }
+    }
+
+    function _performTestFulfillOrderRevertInvalidArrayLength(
+        ConsiderationInterface consideration,
+        Order memory order,
+        bytes memory fulfillOrderCalldata,
+        uint256 itemsLengthFieldOffset,
+        uint256 originalItemsLength,
+        uint256 amtToSubtractFromItemsLength
+    ) internal {
+        _validateOrder(order, consideration);
+
+        bool overwriteItemsLength = amtToSubtractFromItemsLength > 0;
+        if (overwriteItemsLength) {
+            // Get the array length from the calldata and
+            // store the length - amtToSubtractFromItemsLength in the calldata
+            // so that the length value does _not_ accurately represent the actual
+            // total array length.
+            _subtractAmountFromLengthInOrderCalldata(
+                fulfillOrderCalldata,
+                itemsLengthFieldOffset,
+                amtToSubtractFromItemsLength
+            );
+        }
+
+        uint256 finalItemsLength = _getLengthAtOffsetInOrderCalldata(
+            fulfillOrderCalldata,
+            itemsLengthFieldOffset
+        );
+        emit log_named_uint("lengt fooh", finalItemsLength);
```

Comment:
@jameswenzel does this `finalItemsLength` read look reasonable? I'm getting the following emitted: `115792089237316195423570985008687907853269984665640564039457584007913129639935`